src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SecurityChannelFactory.cs

```diff
@@ -0,0 +1,845 @@
+            {
+                get { return _to; }
+            }
+
+            public Uri Via
+            {
+                get { return _via; }
+            }
+
+            protected bool TryGetSecurityFaultException(Message faultMessage, out Exception faultException)
+            {
+                faultException = null;
+                if (!faultMessage.IsFault)
+                {
+                    return false;
+                }
+                MessageFault fault = MessageFault.CreateFault(faultMessage, TransportDefaults.MaxSecurityFaultSize);
+                faultException = SecurityUtils.CreateSecurityFaultException(fault);
+                return true;
+            }
+
+            protected override IAsyncResult OnBeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
+            {
+                throw ExceptionHelper.PlatformNotSupported(); // Issue #31 in progress
+
+                //EnableChannelBindingSupport();
+
+                //return new OpenAsyncResult(this, timeout, callback, state);
+            }
+
+            protected override void OnEndOpen(IAsyncResult result)
+            {
+                throw ExceptionHelper.PlatformNotSupported(); // Issue #31 in progress
+                //OpenAsyncResult.End(result);
+            }
+
+            protected override void OnOpen(TimeSpan timeout)
+            {
+                TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+                EnableChannelBindingSupport();
+
+                SecurityProtocol securityProtocol = SecurityProtocolFactory.CreateSecurityProtocol(
+                    _to,
+                    Via,
+                    null,
+                    typeof(TChannel) == typeof(IRequestChannel),
+                    timeoutHelper.RemainingTime());
+                OnProtocolCreationComplete(securityProtocol);
+                SecurityProtocol.Open(timeoutHelper.RemainingTime());
+                base.OnOpen(timeoutHelper.RemainingTime());
+            }
+
+            protected internal override Task OnOpenAsync(TimeSpan timeout)
+            {
+                return OnOpenAsyncInternal(timeout);
+            }
+
+            private async Task OnOpenAsyncInternal(TimeSpan timeout)
+            {
+                TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+
+                EnableChannelBindingSupport();
+
+                SecurityProtocol securityProtocol = SecurityProtocolFactory.CreateSecurityProtocol(
+                    _to,
+                    Via,
+                    null,
+                    typeof(TChannel) == typeof(IRequestChannel),
+                    timeoutHelper.RemainingTime());
+                OnProtocolCreationComplete(securityProtocol);
+                await SecurityProtocol.OpenAsync(timeoutHelper.RemainingTime());
+                await base.OnOpenAsync(timeoutHelper.RemainingTime());
+            }
+
+            void EnableChannelBindingSupport()
+            {
```

Comment:
What do you think we should do?  I don't want to PNSE here or we defeat the default scenario.  Maybe just stub it all out or remove the call to it?

---
In reply to: [78484932](https://github.com/dotnet/wcf/pull/1515#discussion_r78484932) [](ancestors = 78484932)